nifi-nar-bundles/nifi-extension-utils/nifi-dbcp-base/src/main/java/org/apache/nifi/dbcp/AbstractDBCPConnectionPool.java

```diff
@@ -365,168 +119,143 @@ public List<ConfigVerificationResult> verify(final ConfigurationContext context,
+                .stream()
+                .collect(Collectors.toMap(PropertyDescriptor::getName, s -> {
+                    final PropertyValue propertyValue = context.getProperty(s);
+                    return propertyValue.evaluateAttributeExpressions().getValue();
+                }));
+    }
+
+    protected List<PropertyDescriptor> getDynamicProperties(final ConfigurationContext context) {
+        return context.getProperties()
                 .keySet()
                 .stream()
                 .filter(PropertyDescriptor::isDynamic)
                 .collect(Collectors.toList());
-
-        dynamicProperties.forEach((descriptor) -> {
-            final PropertyValue propertyValue = context.getProperty(descriptor);
-            if (descriptor.isSensitive()) {
-                final String propertyName = StringUtils.substringAfter(descriptor.getName(), SENSITIVE_PROPERTY_PREFIX);
-                dataSource.addConnectionProperty(propertyName, propertyValue.getValue());
-            } else {
-                dataSource.addConnectionProperty(descriptor.getName(), propertyValue.evaluateAttributeExpressions().getValue());
-            }
-        });
-
-        getConnectionProperties(context).forEach(dataSource::addConnectionProperty);
     }
 
-    private KerberosUser getKerberosUser(final ConfigurationContext context) {
-        KerberosUser kerberosUser = null;
-        final KerberosCredentialsService kerberosCredentialsService = context.getProperty(KERBEROS_CREDENTIALS_SERVICE).asControllerService(KerberosCredentialsService.class);
+    protected KerberosUser getKerberosUser(final ConfigurationContext context) {
+        final KerberosUser kerberosUser;
         final KerberosUserService kerberosUserService = context.getProperty(KERBEROS_USER_SERVICE).asControllerService(KerberosUserService.class);
-        final String kerberosPrincipal = context.getProperty(KERBEROS_PRINCIPAL).evaluateAttributeExpressions().getValue();
-        final String kerberosPassword = context.getProperty(KERBEROS_PASSWORD).getValue();
 
         if (kerberosUserService != null) {
             kerberosUser = kerberosUserService.createKerberosUser();
-        } else if (kerberosCredentialsService != null) {
-            kerberosUser = new KerberosKeytabUser(kerberosCredentialsService.getPrincipal(), kerberosCredentialsService.getKeytab());
-        } else if (!StringUtils.isBlank(kerberosPrincipal) && !StringUtils.isBlank(kerberosPassword)) {
-            kerberosUser = new KerberosPasswordUser(kerberosPrincipal, kerberosPassword);
+        } else {
+            kerberosUser = getKerberosUserByCredentials(context);
         }
         return kerberosUser;
     }
 
-    protected String getUrl(ConfigurationContext context) {
-        return context.getProperty(DATABASE_URL).evaluateAttributeExpressions().getValue();
+    protected KerberosUser getKerberosUserByCredentials(final ConfigurationContext context) {
+        return null;
     }
```

Comment:
Instead of having this method implementation return `null` in this abstract class, it looks like it could be removed, and the concrete implementation could override the `getKerberosUser()` method.